server/methods/core/cart-merge.app-test.js

```diff
@@ -90,6 +90,34 @@ describe("Merge Cart function ", function () {
     expect(cart.items.length).to.equal(2);
   });
 
+  it("should increase product quantity if anonymous cart items exists in user's cart before merge", function () {
+    sandbox.stub(Reaction, "getShopId", () => shop._id);
+    let anonymousCart = Factory.create("anonymousCart");
+    let cart = Factory.create("cartOne"); // registered user cart
+    let cartCount = Collections.Cart.find().count();
+    const intialCartQty = cart.items[0].quantity;
+    expect(cartCount).to.equal(2);
+    Collections.Cart.update({
+      "_id": anonymousCart._id, "items._id": anonymousCart.items[0]._id
+    }, { $set: { "items.$.variants._id": cart.items[0].variants_id } });
+    spyOnMethod("mergeCart", cart.userId);
+    const cartRemoveSpy = sandbox.spy(Collections.Cart, "remove");
+    Collections.Cart.update({}, {
+      $set: {
+        sessionId: sessionId
+      }
+    });
+    const mergeResult = Meteor.call("cart/mergeCart", cart._id, sessionId);
+    expect(mergeResult).to.be.ok;
+    anonymousCart = Collections.Cart.findOne(anonymousCart._id);
```

Comment:
I think rather than reusing this variable it might make this more readable to rename this to `anonymousCartAfterMerge` so it's clearer what you are doing